### PR TITLE
[ML] Report correct memory requirements for PyTorch models in stats

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**
The trained models stats API returns `deployment_stats.model_size_bytes`
for deployed PyTorch models. That is intended to report the amount of memory
required for this model to be loaded. This should be the result of
the memory estimation that takes into consideration the doubling of
memory while the process is loading the model plus the process overhead.

Relates #81854

### CODE REVIEW

**Comment by dimitris-athanasiou** [comment]
This is labelled `non-issue` because it address a bug in unreleased versions.

File: docs/reference/ml/trained-models/apis/get-trained-models-stats.asciidoc [hunk 1]
```diff
@@ -116,9 +116,11 @@ The desired number of nodes for model allocation.
 (string)
 include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=model-id]
 
-`model_size`:::
+`model_size_bytes`:::
 (<<byte-units,byte value>>)
-The size of the loaded model in bytes.
+The native memory size required to load this model in bytes.
```
Comment (dimitris-athanasiou) [inline docs/reference/ml/trained-models/apis/get-trained-models-stats.asciidoc:121]: @lcawl Any thoughts on the phrasing here would be appreciated.

Comment (benwtrent) [reply docs/reference/ml/trained-models/apis/get-trained-models-stats.asciidoc:121]: > The native memory size required to load this model in bytes.

This isn't true. We set this also for models loaded in Java. And we should not reference disk size unless specifically referencing PyTorch/native. models

Comment (dimitris-athanasiou) [reply docs/reference/ml/trained-models/apis/get-trained-models-stats.asciidoc:121]: This is the `model_size_stats` under `deployment_stats`. We don't have that for java models.

Comment (davidkyle) [reply docs/reference/ml/trained-models/apis/get-trained-models-stats.asciidoc:121]: It's confusing to have 2 fields called `model_size_bytes` meaning different things. Here we want to express how much free space is required to launch the model.

The motivation is that a user may look at the available memory on the node as reported in the UI and question why a 200MB model cannot be deployed on a node with 300MB of free memory. The answer is that more physical memory is required for loading the model.

Comment (dimitris-athanasiou) [reply docs/reference/ml/trained-models/apis/get-trained-models-stats.asciidoc:121]: Actually, thinking about this, the problem is the name `model_size_bytes`. This should really be something like `estimated_memory_bytes`. @davidkyle Do you think it's worth changing this for 8.0? It'll help us get rid of a bad name going forward.

Comment (droberts195) [reply docs/reference/ml/trained-models/apis/get-trained-models-stats.asciidoc:121]: I don't think we can get away from admitting there are two types of model.  So something like:

```suggestion
If the model is a PyTorch model then this number represents the native memory
size required to load this model in bytes. If the model is a Java model then this
number represents the amount of JVM heap space that the model will consume
in bytes.
```

Maybe not that exact text but it's clear that we're going to have to explain that the meaning varies depending on the type of model.

Comment (dimitris-athanasiou) [reply docs/reference/ml/trained-models/apis/get-trained-models-stats.asciidoc:121]: We discussed this offline and decided we should refactor the way we report this. I'll close this PR and open a different one with the new format.

**Pull request closed without merging**